### ISSUE TITLE
Could you please clarify how many support images are available for each category in the target domain?

### ISSUE THREAD

**Issue by Lzy-dot** [niejiahao1998/IFA#9]
I couldn’t find a description in the paper regarding the number of annotated samples available in the target domain. Could you please clarify this?

**Comment by niejiahao1998** [comment]
We adopt only k images for fine-tuning on the target domain under k-shot. (For example, we randomly pick 1 image as support image for fine-tuning under 1-shot setting)
During inference, it is the same as vanilla few-shot segmentation, namely, using all data for inference.

**Comment by Lzy-dot** [comment]
I understand your point, but in different iterations, different support samples will still be selected for the same category. Does this mean that the labeled data in the target domain is just relatively less compared to the source domain, but it doesn't imply that there are only 1 or 5 support samples per category in the target domain?

**Comment by niejiahao1998** [comment]
Actually, we only pick support image once. That means that we choose 1 image and conduct 20 epoches for fine-tuning (using the same support image).

**Comment by Lzy-dot** [comment]
Oh, I saw it in the dataset_IFA.py and I get it now. Thanks a lot!

**Comment by kkaiwwana** [comment]
Hi there. Thank you for you brilliant work.

I checked the code of both IFA datasets and IFA.py, I don't think I find any code that ensures support images are consistent between different epochs. And as far as I tested, it (both query image and support images) actually varies.

Here is an expamle from isic_IFA.py:
```python
def sample_episode(self, idx):
        class_id = idx % len(self.class_ids)
        class_sample = self.categories[class_id]

        query_name = np.random.choice(self.img_metadata_classwise[class_sample], 1, replace=False)[0]
        support_names = []
        while True:  # keep sampling support set if query == support
            support_name = np.random.choice(self.img_metadata_classwise[class_sample], 1, replace=False)[0]
            if query_name != support_name: support_names.append(support_name)
            if len(support_names) == self.shot: break
```
`idx` is only used to decided the category and then all images are chosen by `random.choice`, without fixed seed. It actually keep yielding different samples. (and therefore disable `shuffle` of dataloader will, of course, not make these stuffs fixed. just in case you think that way.)

It happens in multiple datasets' code, while in some of them that use `idx` to decided the `query_image` (e.g. fss_IFA.py, check codes below), however, the support images seem still pretty random.

```python
def sample_episode(self, idx):
        query_name = self.img_metadata[idx]
        class_sample = self.categories.index(query_name.split('/')[-2])
        if self.split == 'val':
            class_sample += 520
        elif self.split == 'test':
            class_sample += 760
```

Could you @niejiahao1998  provide any clarification to these? I'd love to see it's actually me made something wrong.

**Comment by zhutianlee** [comment]
have you solve this problem? I have same problem
